contracts/utils/contracts/Bitfield.sol

```diff
@@ -26,36 +28,45 @@ library Bitfield {
         0x00000000000000000000000000000000ffffffffffffffffffffffffffffffff;
 
     uint256 internal constant ONE = uint256(1);
-    using Bits for uint256;
+
+    uint256[20] internal BIG_PRIME = [
+        1000003,1000033,1000037,1000039,1000081,1000099,1000117,1000121,1000133,1000151,
+        1000159,1000171,1000183,1000187,1000193,1000199,1000211,1000213,1000231,1000249
+    ];
 
     /**
      * @notice Draws a random number, derives an index in the bitfield, and sets the bit if it is in the `prior` and not
      * yet set. Repeats that `n` times.
      */
-    function randomNBitsFromPrior(
+    function randomNBitsWithPriorCheck(
         uint256 seed,
         uint256[] memory prior,
-        uint256 n
-    ) internal pure returns (uint256[] memory bitfield) {
+        uint256 n,
+        uint256 length
+    ) internal view returns (uint256[] memory bitfield) {
         require(
             n <= countSetBits(prior),
             "`n` must be <= number of set bits in `prior`"
         );
+        require(
+            length < 1000000,
+            "length too large"
+        );
 
         bitfield = new uint256[](prior.length);
+        uint256 prime = BIG_PRIME[seed%20];
+        uint256 begin = uint256(keccak256(abi.encode(seed))) % 1000000 + 1;
         uint256 found = 0;
-        uint256 length = prior.length * 256;
 
         for (uint256 i = 0; found < n; i++) {
-            bytes32 randomness = keccak256(abi.encode(seed + i));
-            uint256 index = uint256(randomness) % length;
+            uint256 index = (prime * (begin + i)) % length;
```

Comment:
what will happen when index > countSetBits(bitfield) ?

Reply:
`countSetBits(prior)` is number of set bits in `prior`, in this case, It is 2/3 of `length`.
`index` is one of bit in `prior`, `index` should < length.
`index > countSetBits(prior)` is possible.

Reply:
so, will it be crashed?

Reply:
No